From 6635e2a399a11c5a454395a966fd417e3530ff97 Mon Sep 17 00:00:00 2001
From: Brian Behlendorf <behlendorf1@llnl.gov>
Date: Sat, 1 Oct 2016 18:33:01 -0700
Subject: [PATCH 1/5] Fix aarch64 type warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Explicitly cast type in splat-rwlock.c test case to silence
the following warning.

  warning: format ‘%ld’ expects argument of type ‘long int’,
  but argument N has type ‘int’

Signed-off-by: Brian Behlendorf <behlendorf1@llnl.gov>
Closes #574
(cherry picked from commit 6c2a66bfa816793fbdcede8854816d526c925574)

Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
---
 module/splat/splat-rwlock.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/module/splat/splat-rwlock.c b/module/splat/splat-rwlock.c
index c17eb07..4576f20 100644
--- a/module/splat/splat-rwlock.c
+++ b/module/splat/splat-rwlock.c
@@ -587,10 +587,10 @@ splat_rwlock_test6(struct file *file, void *arg)
 	if (RWSEM_COUNT(SEM(&rwp->rw_rwlock)) !=
 	    SPL_RWSEM_SINGLE_READER_VALUE) {
 		splat_vprint(file, SPLAT_RWLOCK_TEST6_NAME,
-		             "We assumed single reader rwsem->count "
-			     "should be %ld, but is %ld\n",
-			     SPL_RWSEM_SINGLE_READER_VALUE,
-			     RWSEM_COUNT(SEM(&rwp->rw_rwlock)));
+		    "We assumed single reader rwsem->count "
+		    "should be %ld, but is %ld\n",
+		    (long int)SPL_RWSEM_SINGLE_READER_VALUE,
+		    (long int)RWSEM_COUNT(SEM(&rwp->rw_rwlock)));
 		rc = -ENOLCK;
 		goto out;
 	}
@@ -600,10 +600,10 @@ splat_rwlock_test6(struct file *file, void *arg)
 	if (RWSEM_COUNT(SEM(&rwp->rw_rwlock)) !=
 	    SPL_RWSEM_SINGLE_WRITER_VALUE) {
 		splat_vprint(file, SPLAT_RWLOCK_TEST6_NAME,
-		             "We assumed single writer rwsem->count "
-			     "should be %ld, but is %ld\n",
-			     SPL_RWSEM_SINGLE_WRITER_VALUE,
-			     RWSEM_COUNT(SEM(&rwp->rw_rwlock)));
+		    "We assumed single writer rwsem->count "
+		    "should be %ld, but is %ld\n",
+		    (long int)SPL_RWSEM_SINGLE_WRITER_VALUE,
+		    (long int)RWSEM_COUNT(SEM(&rwp->rw_rwlock)));
 		rc = -ENOLCK;
 		goto out;
 	}
-- 
2.7.4

